### ISSUE TITLE
The binding of joystick buttons  is different from SDL 1.2 to SDL 2.0

### ISSUE THREAD

**Issue by Noxalus** [kripken/emscripten#2193]
I work on a multiplatform project that use SDL 2.0 for other targets than HTML 5 that use SDL 1.2.

But for a reason I don't know, the value of `event.jbutton.button` (for SDL_JOYBUTTONUP or SDL_JOYBUTTONDOWN) is different between SDL 1.2 and SDL 2.0.

Do you know why and where I can found the source code which is causing the problem ?

Thanks in advance.

**Comment by rcgordon** [comment]
I'm bouncing into this issue 3.5 years later, because I just searched for "gamepad" and this is still open:

- SDL joysticks (in 1.2 and 2.0) never make any promises about the order of buttons. The SDL2 gamepad API is what you want if you want to make sure an XBox360 controller always looks like an XBox360 controller, even if it happens to be a PlayStation 3 controller in real life. My assumption is that the Emscripten 1.2 reimplementation just picked different buttons than the SDL2/emscripten port. Since you can pick one or the other for your project, I wouldn't change either implementation at this point.
- SDL2 was ported to Emscripten a while ago, and we're shipping commercial games with it.  :)  This is a tiny plea to not reimplement it for Emscripten, as it's a big API and we are actively maintaining and improving our Emscripten target so you don't have to.  :)

tl;dr: this bug should be closed, imho.

Your friendly neighborhood SDL maintainer,
--ryan.

**Comment by kripken** [comment]
Agreed on those points, and thanks for the help, finding old bugs to close is great.
